### PR TITLE
Feature/506 new download features

### LINKED ISSUE #505: 🐛 BUG: Unknown behavior of download course
Sometimes the download of a course works, sometimes not!
There maybe a problem with the dependences of the zip packages.
Or an other bug we did not discover yet. 

## Additional info:

------
_Please tag this issue if you are sure to which tag(s) it belongs._

### LINKED ISSUE #506: Additional feature request for course download frontend
After the review 26.01.2018 these features should be included for version 0.5 .

## Acceptance criteria:
- [x] Add Error handling if post /download did fail
- [x] Add size checking in frontend instead of backend
- [x] (If the feature above does not work) Add a save state of checked units if the size is negative from backend

### PR DESCRIPTION
**Pull request by PatrickSkowronek**
## Description:
Fixed download course and added some new requested features

Closes #506 
Closes #505 

## Improvements
- Fixed zipping
- Added local check for size of zip
- Add new layout for download

## Known Issues:
_NONE_

------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: api/test/integration/download.ts [hunk 1]
```diff
@@ -53,55 +53,6 @@ describe('DownloadFile', () => {
       res.status.should.be.equal(401);
     });
 
-    it('should fail, no auth', async () => {
```
Comment (aeimer) [inline api/test/integration/download.ts]: Are there any replacements for these deleted tests?

Comment (PatrickSkowronek) [reply api/test/integration/download.ts]: For testcases we would need a real test files. I have an idea for that, but it will take time to automatic test this.

Comment (aeimer) [reply api/test/integration/download.ts]: So then please open a ticket an reference it here. Then I will merge.

**Comment by PatrickSkowronek** [comment]
Edge cases are, files which are less then the limit(currently 50mb) and the complete selection is more then the complete limit(currently 200mb).

File: api/test/integration/download.ts (hunk 1 above)
Comment (PatrickSkowronek) [reply api/test/integration/download.ts]: Ok, I will delete the outdated testcases and open a new ticket.

**Comment by aeimer** [comment]
Ok tested 200mb+, works.
